bin/bump_version.nim

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env -S nim r --verbosity:0 --skipParentCfg:on
+
+## Running this file will:
+## 1. Switch to the `main` branch, if there are no uncommitted changes (exiting
+##    otherwise)
+## 2. Pull the upstream `main` branch from github.com/exercism/configlet
+## 3. Check that the local `main` is not ahead of upstream
+## 4. Create and switch to a new git branch named like `release-x.y.z`
+## 5. Increment the version in `configlet.version` if it exists, otherwise in
+##    the Nimble file (the former is required to support pre-release version
+##    strings like `4.0.0-alpha.36`)
+## 6. Commit the change
+## 7. If `gh` is installed, prompt the user to push to GitHub and create the PR
+## 8. Switch back to the `main` branch
+import std/[os, osproc, strformat, strscans, strutils]
+
+type
+  BumpError = object of CatchableError
+  BumpQuit = object of CatchableError
+
+proc error(msg: string) =
+  ## Prints `msg` and raises a `BumpError`.
+  stderr.writeLine &"Error: {msg}"
+  raise newException(BumpError, "")
+
+proc exit0(msg: string) =
+  ## Prints `msg` and raises a `BumpQuit`.
+  stderr.writeLine msg
+  raise newException(BumpQuit, "")
+
+type
+  Command = enum
+    GitAdd = "git add"
+    GitBranch = "git branch"
+    GitCheckout = "git checkout"
+    GitCommit = "git commit"
+    GitDescribe = "git describe"
+    GitDiff = "git diff"
+    GitLog = "git log"
+    GitLsFiles = "git ls-files"
+    GitPull = "git pull"
+    GitPush = "git push"
+    GitRemote = "git remote"
+    GitRevParse = "git rev-parse"
+    GhPr = "gh pr"
+
+proc exec(command: Command; args: openArray[string] = []): (string, int) =
+  ## Runs the given command with `args`, and returns the output and exit code.
+  var cmd = $command
+  for arg in args:
+    cmd.add ' '
+    cmd.add quoteShell(arg)
+  result = execCmdEx(cmd)
+
+proc execAndCheck(command: Command; args: openArray[string] = [];
+                  errorMsg = ""): string =
+  ## Runs the given command with `args` and returns the output (without a final
+  ## newline).
+  ##
+  ## Raises `BumpError` and prints `errorMsg` if the command's exit code is
+  ## non-zero.
+  var errC = -1
+  (result, errC) = exec(command, args)
+  if errC != 0:
+    stderr.writeLine result
+    if errorMsg.len > 0:
+      error(errorMsg)
+    else:
+      error(&"""the exit code was non-zero: {command} {args.join(" ")}""")
+  stripLineEnd result
+
+proc getGitHubRemoteName(owner, repo: static string): string =
+  ## Returns the name of the remote in that points to `github.com/owner/repo`.
```

Comment:
Should this be:

```suggestion
  ## Returns the name of the remote that points to `github.com/owner/repo`.
```

Reply:
Thanks - fixed in https://github.com/exercism/configlet/pull/529/commits/1705a8dbe4ec9d86b9e4547ab81a4b6d3cf6a0b2.